web/pandas/pdeps/0008-inplace-methods-in-pandas.md

```diff
@@ -0,0 +1,403 @@
+| ``insert``    |
+| ``pop``       |
+| ``update``    |
+| ``isetitem``* |
+
+These methods always operate inplace and don't have the ``inplace`` or ``copy`` keyword.
+
+\* Although ``isetitem`` operates on the original pandas object inplace, it will not change any existing values
+inplace (it will remove the values of the column being set, and insert new values).
+
+**Group 2: Methods that modify the underlying data of the DataFrame/Series object and can be done inplace**
+
+| Method Name     |
+|:----------------|
+| ``where``       |
+| ``fillna``      |
+| ``replace``     |
+| ``mask``        |
+| ``interpolate`` |
+| ``ffill``       |
+| ``bfill``       |
+| ``clip``        |
+
+These methods don't operate inplace by default, but can be done inplace with `inplace=True`. All those methods leave
+the structure of the DataFrame or Series intact (shape, row/column labels), but can mutate some elements of the data of
+the DataFrame or Series.
+
+**Group 3: Methods that modify the DataFrame/Series object, but not the pre-existing values**
+
+| Method Name                 | Keyword               |
+|:----------------------------|-----------------------|
+| ``drop`` (dropping columns) | ``inplace``           |
+| ``rename``                  | ``inplace``, ``copy`` |
+| ``rename_axis``             | ``inplace``, ``copy`` |
+| ``reset_index``             | ``inplace``           |
+| ``set_index``               | ``inplace``           |
+| ``astype``                  | ``copy``              |
+| ``infer_objects``           | ``copy``              |
+| ``set_axis``                | ``copy``              |
+| ``set_flags``               | ``copy``              |
+| ``to_period``               | ``copy``              |
+| ``to_timestamp``            | ``copy``              |
+| ``tz_localize``             | ``copy``              |
+| ``tz_convert``              | ``copy``              |
+| ``Series.swaplevel``*       | ``copy``              |
+| ``concat``                  | ``copy``              |
+
+\* The `copy` keyword is only available for `Series.swaplevel` and not for `DataFrame.swaplevel`.
+
+These methods can change the structure of the DataFrame or Series, such as changing the shape by adding or removing
+columns, or changing the row/column labels (changing the index/columns attributes), but don't modify the existing
+underlying data of the object.
+
+All those methods (except for `set_flags`) make a copy of the full data by default, but can be performed inplace with
+avoiding copying all data (currently enabled with the `inplace` or `copy` keyword).
+
+Some of these methods only have a `copy` keyword instead of an `inplace`
+keyword. These allow the user to avoid a copy, but don't update the original object inplace and instead return a
+new object referencing the same data.
+
+Two methods also have both keywords: `rename`, `rename_axis`, with the `inplace` keyword overriding `copy`.
+
+**Group 4: Methods that can never operate inplace**
```

Comment:
"Methods that can never operate inplace despite presence of `inplace` or `copy` keywords`